packages/node/src/typescript.ts

```diff
@@ -486,7 +487,11 @@ export function fixConfig(
 
   // If not specified, the default Node.js module is CommonJS.
   if (config.compilerOptions.module === undefined) {
-    config.compilerOptions.module = 'CommonJS';
+    config.compilerOptions.module = isEsm ? 'nodenext' : 'CommonJS';
```

Comment:
We have to be very careful when changing the default here. Last time it was changed, we broke a bunch of customers.

See https://github.com/vercel/vercel/pull/7954#issuecomment-1374878892

Reply:
Are you saying that `isEsm` should default to `false`? And that we should slap a big ol' warning comment on `isEsm`?

Reply:
`isEsm` is only set when `fixConfig()` is called by `startDevServer()` (e.g. `vc dev`). The intention was to not touch `vc deploy` or `vc build` code paths. `isEsm` will be true only if `vc dev` is executing an ESM function based https://github.com/vercel/vercel/blob/46348201b4eadb03b403f99dc5574c31d2b3926e/packages/node/src/index.ts#L554-L557.

Reply:
@cb1kenobi If this is only used for development, I think it should happen in `startDevServer()` only since `fixConfig()` is used for both dev and prod. I dont want someone to see `isEsm` in the future and think they need to pass that variable for prod too.